src/Ng/Lightning.cs

```diff
@@ -0,0 +1,397 @@
+            Console.WriteLine("entries.");
+        }
+
+        public static void Run(string[] args, CancellationToken cancellationToken)
+        {
+            PrintLightning();
+
+            IDictionary<string, string> arguments = CommandHelpers.GetArguments(args, 1);
+            if (arguments == null || arguments.Count == 0)
+            {
+                PrintUsage();
+                return;
+            }
+
+            string command = CommandHelpers.Get(arguments, "command");
+            if (string.IsNullOrEmpty(command))
+            {
+                PrintUsage();
+                return;
+            }
+
+            bool verbose = CommandHelpers.GetVerbose(arguments);
+
+            var log = verbose ? Console.Out : new StringWriter();
+
+            switch (command.ToLowerInvariant())
+            {
+                case "charge":
+                case "prepare":
+                    PrepareAsync(log, arguments).Wait(cancellationToken);
+                    break;
+                case "strike":
+                    StrikeAsync(log, arguments).Wait(cancellationToken);
+                    break;
+                default:
+                    PrintUsage();
+                    break;
+            }
+        }
+
+        private static async Task PrepareAsync(TextWriter log, IDictionary<string, string> arguments)
+        {
+            // Read arguments
+            string outputFolder = CommandHelpers.Get(arguments, "outputFolder");
+            string catalogIndex = CommandHelpers.Get(arguments, "catalogIndex");
+            string templateFile = CommandHelpers.Get(arguments, "templateFile");
+            string batchSize = CommandHelpers.Get(arguments, "batchSize");
+            string storageAccount = CommandHelpers.Get(arguments, "storageAccount");
+            string container = CommandHelpers.Get(arguments, "container");
+            string baseAddress = CommandHelpers.Get(arguments, "baseAddress");
+
+            if (string.IsNullOrEmpty(outputFolder)
+                || string.IsNullOrEmpty(catalogIndex)
+                || string.IsNullOrEmpty(templateFile)
+                || string.IsNullOrEmpty(batchSize) 
+                || string.IsNullOrEmpty(storageAccount)
+                || string.IsNullOrEmpty(container)
+                || string.IsNullOrEmpty(baseAddress))
+            {
+                PrintUsage();
+                return;
+            }
+
+            log.WriteLine("Making sure folder {0} exists.", outputFolder);
+            if (!Directory.Exists(outputFolder))
+            {
+                Directory.CreateDirectory(outputFolder);
+            }
+
+            // Create reindex file
+            log.WriteLine("Start preparing lightning reindex file...");
+
+            var latestCommit = DateTime.MinValue;
+            int numberOfEntries = 0;
+            string indexFile = Path.Combine(outputFolder, "index.txt");
```

Comment:
You are essentially introducing an alternative "index" structure to the catalog's existing index structure. We could consolidate on a single structure this should [theoretically] reduce the overall complexity because we'd just have one model for iteration, one model for cursors etc.

However, I don't think there is an urgent need for this consolidation. So not really a code-review comment but something we should discuss and experiment with perhaps. In the process of doing that perhaps we could also look at a couple of other things in the catalog - for example the read batch size could be independent of the write batch commit.